pkg/rtc/participant.go

```diff
@@ -1242,12 +1249,12 @@ func (p *ParticipantImpl) onMediaTrack(track *webrtc.TrackRemote, rtpReceiver *w
 		p.params.Logger.Warnw("no permission to publish mediaTrack", nil,
 			"source", publishedTrack.Source(),
 		)
+		p.removePublishedTrack(publishedTrack)
```

Comment:
Adding this in the publish path. That's the reason for splitting out the `remotePublishedTrack` method ☝️ .

I thought about adding the check before adding the track and not adding it. But, it is already published, so adding it and removing felt like the right flow and also the fact that things could race (i. e. permissions could be changing when a published track is being added, note that there are probably races still, for e.g. rapidly changing permissions).